code/modules/mob/inventory.dm

```diff
@@ -350,24 +350,34 @@
 		M.s_active.handle_item_insertion(src)
 		return TRUE
 
-	var/obj/item/storage/S = M.get_inactive_held_item()
-	if(istype(S) && S.can_be_inserted(src,1))	//see if we have box in other hand
-		S.handle_item_insertion(src)
-		return TRUE
+	var/obj/item/S = M.get_inactive_held_item()
+	var/datum/component/storage/STR
+	if(istype(S))
+		STR = S.GetComponent(/datum/component/storage)
+		if(STR.can_be_inserted(src,TRUE))	//see if we have box in other hand

```

Comment:
these should be signals IMO, switching something to a datum component and then not using signals makes the API abstractions pointless

Reply:
Especially when you're constantly repeating this pattern